### PR TITLE
[4055] Added scrollbar to Dropdown, and open upwards feature

### PR DESCRIPTION
**Pull request by pablobentivengo**
**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/4055

**Problem:**
* If there are more than 4 options for the person to choose, the dropdown should have a vertical scroll.
* If the dropdown is closer to the end of the screen, it should open going up

**In this PR:**
* Modified FieldSelect.style adding scrollbar width by 10px, and set the scrollbar display to auto. Added an option (_openUpDirection) which opens the dropdown upwards in case it is needed.
* Modified FieldSelect.container added handleOpenDirection() which manages if the dropdown should open up or down.
* Modified FieldSelect.component to use openUpDirection prop to render as a mod option in renderOptions()

### CODE REVIEW

**Comment by aleksandrakorolova** [comment]
1. No upper bound if dropdown opens up. Please, add upper bound
Screenshot 2022-01-13 at 08 15 37
2. Scroll appears and immediately disappears when opening dropdown with less than 5 options. Please, don't show scroll bar if there are less than 5 options
https://drive.google.com/file/d/1nmuafeo8L0fxsXwGJOBvdlQaSocElvU7/view?usp=sharing

**Comment by aleksandrakorolova** [comment]
1. Fixed
2. Still, can reproduce for Currency switcher 
https://drive.google.com/file/d/1XWP15_Lv6vE16beuufuzKQUkYY9CUTUp/view?usp=sharing

File: packages/scandipwa/src/component/FieldSelect/FieldSelect.style.scss [hunk 1]
```diff
@@ -73,27 +73,52 @@ $select-arrow-width: 14px !default;
 
         // Firefox support
         /* stylelint-disable-next-line declaration-no-important */
-        scrollbar-width: none !important;
+        scrollbar-width: 10px !important;
         /* stylelint-disable-next-line declaration-no-important */
-        scrollbar-color: none !important;
-
+        scrollbar-color: gray !important;
+
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/FieldSelect/FieldSelect.style.scss:79]: Remove this indent

File: packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js [hunk 2]
```diff
@@ -128,6 +136,7 @@ export class FieldSelectContainer extends PureComponent {
         if (!this.isSelectDisabled()) {
             this.setState(({ isExpanded }) => ({ isExpanded: !isExpanded }));
         }
+        this.handleOpenDirection();
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js:139]: Add white space after if block

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]
Looks cool!
Just some minor changes required.

File: packages/scandipwa/src/component/FieldSelect/FieldSelect.style.scss (hunk 1 above)
Comment (pablobentivengo) [reply packages/scandipwa/src/component/FieldSelect/FieldSelect.style.scss]: Done!

File: packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js (hunk 2 above)
Comment (pablobentivengo) [reply packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js:140]: Done!

File: packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js [hunk 3]
```diff
@@ -236,6 +246,30 @@ export class FieldSelectContainer extends PureComponent {
         }
     }
 
+    handleOpenDirection() {
+        const windowHeight = document.documentElement.clientHeight;
+        const rect = this.fieldRef.getBoundingClientRect();
+        const bottomPosition = Math.round(windowHeight - rect.bottom);
+        const minimunDistance = 200;
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js:253]: Add please this as constant and make more descriptive name

File: packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js [hunk 4]
```diff
@@ -236,6 +246,30 @@ export class FieldSelectContainer extends PureComponent {
...
+        if (bottomPosition < minimunDistance) {
+            this.setState({ isUpDirection: true });
+        } else {
+            this.setState({ isUpDirection: false });
+        }
+    }
+
+    handleScrollbar() {
+        const options = this.getOptions();
+        const minimunOptions = 5;
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js:264]: Same here

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js (hunk 3 above)
Comment (pablobentivengo) [reply packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js]: Done!

File: packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js (hunk 4 above)
Comment (pablobentivengo) [reply packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js]: Done!

File: packages/scandipwa/src/component/FieldSelect/FieldSelect.config.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+export const DROPDOWNLIST_CONFIG = {
+    scrollMinItems: 5,
+    minimunHeight: 200
+};
+
+export default DROPDOWNLIST_CONFIG;
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/FieldSelect/FieldSelect.config.js:17]: Split it to 2 separate configs consts

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

**Review by AleksandrsKondratjevs — APPROVED** [review]

**Pull request merged**